internal/datastore/mongo_store.go

```diff
@@ -46,35 +47,63 @@ func NewMongoDB() *MongoDB {
 	return &new_mongodb
 }
 
-func (mongodb MongoDB) CreateQueue(queue models.Queue) models.QueueId {
+func (mongodb MongoDB) CreateQueue(queue models.Queue) (models.QueueId, error) {
 	// Set id to empty so its generated by mongoDB
 	queue.Id = ""
 
 	result, err := mongodb.Queue.InsertOne(context.TODO(), queue)
 
 	if err != nil {
-		log.Fatal(err)
+		return queue.Id, err
 	}
 
 	stringId := result.InsertedID.(primitive.ObjectID).Hex()
 
-	return models.QueueId(stringId)
+	return models.QueueId(stringId), nil
 }
 
-func (mongodb MongoDB) ReadQueue(models.QueueId) models.Queue {
-	panic("Not implemented")
-}
+func (mongodb MongoDB) ReadQueue(id models.QueueId) (models.Queue, error) {
+	result := models.Queue{}
 
-func (mongodb MongoDB) PauseQueue(models.QueueId) {
-	panic("Not implemented")
+	// convert id string to ObjectId
+	queueId, _ := primitive.ObjectIDFromHex(string(id))
+
+	err := mongodb.Queue.FindOne(context.TODO(), bson.M{"_id": queueId}).Decode(&result)
+
+	if err != nil {
+		return result, err
+	}
+
+	return result, nil
 }
 
-func (mongodb MongoDB) ResumeQueue(models.QueueId) {
-	panic("Not implemented")
+func (mongodb MongoDB) SetIsPaused(id models.QueueId, isPaused bool) (*mongo.UpdateResult, error) {
```

Comment:
I don't think we should return a type dependent on mongo here. `QueueStore` interface should be independent of the database we are using.
We can instead return a string and a boolean. The string describing the error and boolean true if successful and false if there is an error.

Reply:
Dependency removed. Although did not see any use of returning boolean flag. Just returning error for now. Plz review